packages/twenty-server/src/engine/metadata-modules/remote-server/remote-table/remote-table-schema-update/remote-table-schema-update.service.ts

```diff
@@ -0,0 +1,176 @@
+import { Injectable } from '@nestjs/common';
+
+import { getForeignTableColumnName as convertToForeignTableColumnName } from 'src/engine/metadata-modules/remote-server/remote-table/foreign-table/utils/get-foreign-table-column-name.util';
+import { DistantTables } from 'src/engine/metadata-modules/remote-server/remote-table/distant-table/types/distant-table';
+import {
+  RemoteTableStatus,
+  DistantTableUpdate,
+} from 'src/engine/metadata-modules/remote-server/remote-table/dtos/remote-table.dto';
+import { RemoteTableEntity } from 'src/engine/metadata-modules/remote-server/remote-table/remote-table.entity';
+import { fetchTableColumns } from 'src/engine/metadata-modules/remote-server/remote-table/utils/fetch-table-columns.util';
+import { PostgresTableSchemaColumn } from 'src/engine/metadata-modules/remote-server/types/postgres-table-schema-column';
+import {
+  WorkspaceMigrationColumnAction,
+  WorkspaceMigrationColumnCreate,
+  WorkspaceMigrationColumnActionType,
+  WorkspaceMigrationColumnDrop,
+} from 'src/engine/metadata-modules/workspace-migration/workspace-migration.entity';
+import { WorkspaceDataSourceService } from 'src/engine/workspace-datasource/workspace-datasource.service';
+
+@Injectable()
+export class RemoteTableSchemaUpdateService {
+  constructor(
+    private readonly workspaceDataSourceService: WorkspaceDataSourceService,
+  ) {}
+
+  public async getDistantTablesWithUpdates({
+    remoteServerSchema,
+    workspaceId,
+    remoteTables,
+    distantTables,
+  }: {
+    remoteServerSchema: string;
+    workspaceId: string;
+    remoteTables: RemoteTableEntity[];
+    distantTables: DistantTables;
+  }) {
+    const schemaPendingUpdates =
+      await this.getSchemaUpdatesBetweenForeignAndDistantTables({
+        workspaceId,
+        remoteTables,
+        distantTables,
+      });
+
+    const remoteTablesDistantNames = remoteTables.map(
```

Comment:
small detail: Could use a set for faster search later
```typescript
 const remoteTablesDistantNames = new Set(remoteTables.map(
    (remoteTable) => remoteTable.distantTableName
  ));
```
where you can lookup with `remoteTablesDistantNames.has(tableName)` instead of includes